SDK_dotnet/GluwaClient.cs

```diff
@@ -0,0 +1,402 @@
+﻿using Nethereum.ABI;
+using Nethereum.Signer;
+using SDK_dotnet.Models;
+using SDK_dotnet.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SDK_dotnet
+{
+    public sealed class GluwaClient
+    {
+        private readonly bool mIsDevEnv;
+        private readonly string mApiKey;
+        private readonly string mSecretKey;
+        private readonly string mAddress;
+        private readonly string mPrivateKey;
+        private readonly string mBaseUrl;
+        private readonly string mWebhookSecretKey;
+
+        public GluwaClient(
+            string apiKey,
+            string secretKey,
+            string address,
+            string privateKey,
+            bool isDevEnv,
+            string webhookSecretKey
+            )
+        {
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                throw new ArgumentNullException(nameof(apiKey));
+            }
+            else if (string.IsNullOrEmpty(secretKey))
+            {
+                throw new ArgumentNullException(nameof(secretKey));
+            }
+            else if (string.IsNullOrEmpty(address))
+            {
+                throw new ArgumentNullException(nameof(address));
+            }
+            else if (string.IsNullOrEmpty(privateKey))
+            {
+                throw new ArgumentNullException(nameof(privateKey));
+            }
+
+            mApiKey = apiKey;
+            mSecretKey = secretKey;
+            mAddress = address;
+            mPrivateKey = privateKey;
+            mIsDevEnv = isDevEnv;
+            mWebhookSecretKey = webhookSecretKey;
+
+            if (mIsDevEnv)
+            {
+                mBaseUrl = "https://sandbox.api.gluwa.com";
+            }
+            else
+            {
+                mBaseUrl = "https://api.gluwa.com";
+            }
+        }
+
+        public async Task<Result<BalanceResponse>> GetAddressAsync(ECurrency? currency)
+        {
+            var result = new Result<BalanceResponse>();
+            string requestUri = $"{mBaseUrl}/v1/{currency}/Addresses/{mAddress}";
+
+            using (HttpClient httpClient = new HttpClient())
+            {
+                HttpResponseMessage message = await httpClient.GetAsync(requestUri);
+                if (message.IsSuccessStatusCode)
+                {
+                    BalanceResponse balanceResponse = await message.Content.ReadAsAsync<BalanceResponse>();
+                    result.IsSuccess = true;
+                    result.Data = balanceResponse;
+
+                    return result;
+                }
+                else
+                {
+                    throw new HttpListenerException((int)message.StatusCode, message.ReasonPhrase);
```

Comment:
hmmm not a good way to handle errors. This is public client, which means that if there is any error responses coming back from the server, the user needs to read the error code or a message to figure out what went wrong. This way, the user won't know what happened. More to be discussed in a separately on slack

Reply:
Returns an int code and a message. Because the code(Badrequest for example) comes with the message. Is that correct?

Reply:
Seems to be an outdated question (looking at the current code, it seems it's setup correctly) so closing this